extensions/measurement-tracking/src/panels/PanelMeasurementTableTracking/index.js

```diff
@@ -129,12 +143,104 @@ function PanelMeasurementTableTracking({ servicesManager, extensionManager }) {
+    const { referenceSeriesUID, SOPInstanceUID } = measurement;
+
+    const displaySets = DisplaySetService.getDisplaySetsForSeries(referenceSeriesUID);
+    const displaySet = displaySets.find(ds => {
+      return ds.images && ds.images.some(i => i.SOPInstanceUID === SOPInstanceUID)
+    });
+
+    const imageIndex = displaySet.images.map(i => i.SOPInstanceUID).indexOf(SOPInstanceUID);
+
+    viewportGridService.setDisplaysetForViewport({
+      viewportIndex: viewportGrid.activeViewportIndex,
+      displaySetInstanceUID: displaySet.displaySetInstanceUID,
+      imageIndex
+    });
+
+    onMeasurementItemClickHandler({ id, isActive });
+  };
+
+  const onMeasurementItemEditHandler = ({ id }) => {
+    const measurement = MeasurementService.getMeasurement(id);
+
+    let dialogId;
+    const onSubmitHandler = ({ action, value }) => {
+      switch (action.id) {
+        case 'save': {
+          MeasurementService.update(id, {
+            ...measurement,
+            ...value
+          });
+          UINotificationService.show({
+            title: 'Measurements',
+            message: 'Label updated successfully',
+            type: 'success'
+          });
+        }
+      }
+      UIDialogService.dismiss({ id: dialogId });
+    };
+    dialogId = UIDialogService.create({
+      centralize: true,
+      isDraggable: false,
+      useLastPosition: false,
+      showOverlay: true,
+      content: Dialog,
+      contentProps: {
+        title: 'Enter label for your annotation:',
+        noCloseButton: true,
+        value: { label: measurement.label || '' },
+        body: ({ value, setValue }) => {
+          const onChangeHandler = () => setValue(value => ({
+            ...value, label: event.target.value
+          }));
+          const onKeyPressHandler = event => {
+            if (event.key === 'Enter') {
+              onSubmitHandler({ value, action: { id: 'save' } });
+            }
+          };
+          return (
+            <div className="p-4 bg-primary-dark">
+              <Input
+                autoFocus
+                className="mt-2 bg-black border-primary-main"
+                type="text"
+                containerClassName="mr-2"
+                value={value.label}
+                onChange={onChangeHandler}
+                onKeyPress={onKeyPressHandler}
+              />
+            </div>
+          );
+        },
+        actions: [
+          { id: 'cancel', text: 'Cancel', type: 'secondary' },
+          { id: 'save', text: 'Save', type: 'primary' },
+        ],
+        onSubmit: onSubmitHandler
+      }
+    });
+  };
+
+  const onMeasurementItemClickHandler = ({ id, isActive }) => {
+    if (!isActive) {
+      const measurements = [...displayMeasurements];
+      const measurement = measurements.find(m => m.id === id);
+      measurements.forEach(m => m.isActive = m.id !== id ? false : true);
+      measurement.isActive = true;
+      setDisplayMeasurements(measurements);
+    }
+  };
```

Comment:
👀 I feel like there has to be a better way to do this. Follow-up issue?

Reply:
Yep, we need a centralized measurement state, if i click in a measurement it should be active in the measurement panel, currently its not reactive at all.